pkg/authorization/api/conversion_test.go

```diff
@@ -0,0 +1,352 @@
+			f: func() error {
+				return Convert_api_ClusterRoleBinding_To_rbac_ClusterRoleBinding(&ClusterRoleBinding{
+					Subjects: []api.ObjectReference{
+						{Kind: "fancyuser"},
+					},
+				}, &rbac.ClusterRoleBinding{}, nil)
+			},
+		},
+		{
+			name:     "invalid RBAC subject kind",
+			expected: `invalid kind for rbac subject: "evenfancieruser"`,
+			f: func() error {
+				return Convert_rbac_ClusterRoleBinding_To_api_ClusterRoleBinding(&rbac.ClusterRoleBinding{
+					Subjects: []rbac.Subject{
+						{Kind: "evenfancieruser"},
+					},
+				}, &ClusterRoleBinding{}, nil)
+			},
+		},
+	} {
+		if err := test.f(); err == nil || test.expected != err.Error() {
+			t.Errorf("%s failed: expected %q got %v", test.name, test.expected, err)
+		}
+	}
+}
+
+var fuzzer = fuzz.New().NilChance(0).Funcs(
+	func(*unversioned.TypeMeta, fuzz.Continue) {}, // Ignore TypeMeta
+	func(*runtime.Object, fuzz.Continue) {},       // Ignore AttributeRestrictions since they are deprecated
+	func(ocrb *ClusterRoleBinding, c fuzz.Continue) {
+		c.FuzzNoCustom(ocrb)
+		setRandomOriginRoleBindingData(ocrb.Subjects, &ocrb.RoleRef, "", c)
+	},
+	func(orb *RoleBinding, c fuzz.Continue) {
+		c.FuzzNoCustom(orb)
+		setRandomOriginRoleBindingData(orb.Subjects, &orb.RoleRef, orb.Namespace, c)
+	},
+	func(rcrb *rbac.ClusterRoleBinding, c fuzz.Continue) {
+		c.FuzzNoCustom(rcrb)
+		setRandomRBACRoleBindingData(rcrb.Subjects, &rcrb.RoleRef, "", c)
+	},
+	func(rrb *rbac.RoleBinding, c fuzz.Continue) {
+		c.FuzzNoCustom(rrb)
+		setRandomRBACRoleBindingData(rrb.Subjects, &rrb.RoleRef, rrb.Namespace, c)
+	},
+	func(rr *rbac.Role, c fuzz.Continue) {
+		c.FuzzNoCustom(rr)
+		sortAndDeduplicateRBACRulesFields(rr.Rules) // []string <-> sets.String
+	},
+	func(rcr *rbac.ClusterRole, c fuzz.Continue) {
+		c.FuzzNoCustom(rcr)
+		sortAndDeduplicateRBACRulesFields(rcr.Rules) // []string <-> sets.String
+	},
+)
+
+func setRandomRBACRoleBindingData(subjects []rbac.Subject, roleRef *rbac.RoleRef, namespace string, c fuzz.Continue) {
+	for i := range subjects {
+		subject := &subjects[i]
+		subject.APIVersion = rbac.GroupName
+		setValidRBACKindAndNamespace(subject, i, c)
+	}
+	roleRef.APIGroup = rbac.GroupName
+	roleRef.Kind = getKind(namespace)
+}
+
+func setValidRBACKindAndNamespace(subject *rbac.Subject, i int, c fuzz.Continue) {
+	kinds := []string{rbac.UserKind, rbac.GroupKind, rbac.ServiceAccountKind}
+	kind := kinds[c.Intn(len(kinds))]
+	subject.Kind = kind
+
+	if subject.Kind != rbac.ServiceAccountKind {
+		subject.Namespace = ""
+	}
+
+	switch subject.Kind {
+
+	case rbac.UserKind:
+		if len(uservalidation.ValidateUserName(subject.Name, false)) != 0 {
+			subject.Name = fmt.Sprintf("validusername%d", i)
+		}
+
+	case rbac.GroupKind:
+		if len(uservalidation.ValidateGroupName(subject.Name, false)) != 0 {
+			subject.Name = fmt.Sprintf("validgroupname%d", i)
+		}
+
+	case rbac.ServiceAccountKind:
+		if len(validation.ValidateNamespaceName(subject.Namespace, false)) != 0 {
```

Comment:
RBAC service account references aren't required to have namespaces.  Some shouldn't have them.